### PR TITLE
更新至 G 最新版本

### PR DESCRIPTION
**Pull request by xiaoiver**
### 👀 PR includes

更新到 G 最新版本，包括 `@antv/g-canvas` 等相关依赖。

✨ Feature

- [ ] New feature

🎨 Enhance

- [ ] Code style optimization
- [ ] Refactoring
- [ ] Change the UI
- [ ] Improve the performance
- [ ] Type optimization

🐛 Bugfix

- [ ] Solve the issue and close #0

🔧 Chore

- [ ] Test case
- [ ] Docs / demos update
- [ ] CI / workflow
- [ ] Release version
- [ ] Other ()

### 🖼️ Screenshot

| Before | After |
| ------ | ----- |
| ❌      | ✅     |

### 🔍 Self-Check before the merge

- [ ] Add or update relevant docs.
- [ ] Add or update relevant demos.
- [ ] Add or update test case.
- [ ] Add or update relevant TypeScript definitions.

### CODE REVIEW

File: packages/s2-react/playground/components/ChartSheet.tsx [hunk 1]
```diff
@@ -44,6 +44,8 @@ const onDataCellRender: SheetComponentsProps['onDataCellRender'] = (cell) => {
 
   // https://g2.antv.antgroup.com/manual/extra-topics/bundle#g2stdlib
   renderToMountedElement(chartOptions, {
+    // @ts-ignore
+    // FIXME: After upgrading @antv/g2, the type of group should be Group
```
Comment (xiaoiver) [inline packages/s2-react/playground/components/ChartSheet.tsx:48]: G2 升级版还没发，所以会有 5.0 -> 6.0 类型不兼容问题，发布后需要移除。

**Comment by lcx-seima** [comment]
有没有必要把 g 提出为 peer dependency

**Review by wjgogogo — APPROVED** [review]

File: packages/s2-core/__tests__/unit/common/icons/gui-icon-spec.ts [hunk 2]
```diff
@@ -96,7 +96,7 @@ describe('GuiIcon Tests', () => {
     });
 
     const spy = jest.spyOn(icon, 'getImage');
-    const oldVal = icon.iconImageShape.style.img;
+    const oldVal = icon.iconImageShape.style.src;
```
Comment (lijinke666) [inline packages/s2-core/__tests__/unit/common/icons/gui-icon-spec.ts:99]: 对于 S2 只有这一个 `img => src` 的 break change 吗?

**Comment by lijinke666** [comment]
单测挂了, 看起来是升级导致的

image

**Pull request merged**